### PR TITLE
New Script: CheckMk

### PR DESCRIPTION
**Pull request by michelroegl-brunner**
> **🛠️ Note:**  
> We are meticulous about merging code into the main branch, so please understand that pull requests not meeting the project's standards may be rejected. It's never personal!  
> 🎮 **Note for game-related scripts:** These have a lower likelihood of being merged.

---

## ✍️ Description

Adds new Script for CheckMK. Can be tested in the develop branch.

 

- - -
**_Please remove unneeded lines!_**
- Related Issue: #674 

---

## 🛠️ Type of Change
Please check the relevant options:  
- [ ] Bug fix (non-breaking change that resolves an issue)  
- [ ] New feature (non-breaking change that adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to change unexpectedly)  
- [X] New script (a fully functional and thoroughly tested script or set of scripts)  

---

## ✅ Prerequisites
The following steps must be completed for the pull request to be considered:  
- [X] Self-review performed (I have reviewed my code to ensure it follows established patterns and conventions.)  
- [X] Testing performed (I have thoroughly tested my changes and verified expected functionality.)  
- [ ] Documentation updated (I have updated any relevant documentation)

---

### CODE REVIEW

File: install/checkmk-install.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+msg_info "Installing Dependencies"
+$STD apt-get install -y \
+  curl \
+  mc 
+msg_ok "Installed Dependencies"
+
+msg_info "Install Checkmk"
+RELEASE=$(curl -fsSL https://api.github.com/repos/checkmk/checkmk/tags | grep "name" | awk '{print substr($2, 3, length($2)-4) }' | grep -v "*-rc" | tail -n +2 | head -n 1)
+wget -q https://download.checkmk.com/checkmk/${RELEASE}/check-mk-raw-${RELEASE}_0.bookworm_amd64.deb -O /opt/checkmk.deb
+$STD apt-get install -y /opt/checkmk.deb
```
Comment (MickLesk) [inline install/checkmk-install.sh:26]: apt install or dpkg -i?

File: ct/checkmk.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  check_container_storage
+  check_container_resources
+  if [[ ! -f /opt/checkmk_version.txt ]]; then
+    msg_error "No ${APP} Installation Found!"
+    exit
+  fi
+  RELEASE=$(curl -fsSL https://api.github.com/repos/checkmk/checkmk/tags | grep "name" | awk '{print substr($2, 3, length($2)-4) }' | grep -v "*-rc" | tail -n +2 | head -n 1)
+  if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
+    msg_info "Updating ${APP} to v${RELEASE}"
+    omd stop monitoring &>/dev/null
```
Comment (MickLesk) [inline ct/checkmk.sh:35]: wtf is omd? :D

File: install/checkmk-install.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies"
+$STD apt-get install -y \
+  curl \
+  mc 
+msg_ok "Installed Dependencies"
```
Comment (MickLesk) [inline install/checkmk-install.sh:21]: sudo missing

File: ct/checkmk.sh (hunk 2 above)
Comment (michelroegl-brunner) [reply ct/checkmk.sh:35]: omd is a checkmk internal managmetn tool. It gets installed with it^^

File: install/checkmk-install.sh (hunk 1 above)
Comment (michelroegl-brunner) [reply install/checkmk-install.sh:26]: The guid says apt install, nut i can change it if you like.

cmk

Comment (MickLesk) [reply install/checkmk-install.sh:27]: if it works, try it

Comment (michelroegl-brunner) [reply install/checkmk-install.sh]: Changed and it is working

Comment (MickLesk) [reply install/checkmk-install.sh]: can you execute: apt-get -f install after dpkg -i ? (not in the script, in the finished lxc). If something installed, we need apt-get install ^^

Comment (michelroegl-brunner) [reply install/checkmk-install.sh]: got nothing.
image

**Review by MickLesk — APPROVED** [review]

**Pull request merged**